source/credentials.c

```diff
@@ -199,8 +207,17 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
 
     environment_provider = aws_credentials_provider_new_environment(allocator, &environment_options);
 
+    bool success = true;
```

Comment:
It would be simpler (and less error prone) if `bool success = false` by default, and we only set it true in one place.

instead of needing to remember to set it false in the many places an error might occur

Reply:
Agreed